pkg/splunk/enterprise/configuration.go

```diff
@@ -377,7 +381,10 @@ func getSplunkStatefulSet(ctx context.Context, client splcommon.ControllerClient
 	// prepare misc values
 	ports := splcommon.SortContainerPorts(getSplunkContainerPorts(instanceType)) // note that port order is important for tests
 	annotations := splcommon.GetIstioAnnotations(ports)
-	selectLabels := getSplunkLabels(cr.GetName(), instanceType, spec.ClusterMasterRef.Name)
+	selectLabels := getSplunkLabels(cr.GetName(), instanceType, spec.ClusterManagerRef.Name)
+	if len(spec.ClusterMasterRef.Name) > 0 && len(spec.ClusterManagerRef.Name) == 0 {
```

Comment:
How will this work for old CRs which do not have ClusterManagerRef? OR should there be an `||` here?

Reply:
Good point, I think this could eventually cause problems because it sets the Manager as default.
I inverted the order in the latest commit so Master continues to be the default, and the labels are updated afterward if Manager is set.